R/prepare_matches_mapper.R

```diff
@@ -6,16 +6,24 @@
 #' @return A dataframe that contains information on multiple matches
 #' @noRd
 prepare_matches_mapper <- function(mapper, activities) {
+  stopifnot(is.logical(mapper$multi_match))
   activities <- activities |>
     select("activity_uuid_product_uuid", "activity_name", "reference_product_name", "unit")
 
-  mapper |>
-    left_join(activities, by = "activity_uuid_product_uuid") |>
-    summarise(across(everything(), ~ paste0(na.omit(unique(.)), collapse = "; ")), .by = c("country", "main_activity", "clustered")) |>
-    ungroup() |>
+  mapper_df <- mapper |>
+    left_join(activities, by = "activity_uuid_product_uuid")
+  stop_if_some_columns_have_more_than_one_unique_value(mapper_df)
+
+  non_summarised_df <- mapper_df |>
+    select(-c("activity_uuid_product_uuid", "activity_name")) |>
+    summarise(across(everything(), \(x) unique(x)), .by = c("country", "main_activity", "clustered"))
+  summarised_df <- mapper_df |>
+    select(c("country", "main_activity", "clustered", "activity_uuid_product_uuid", "activity_name")) |>
+    summarise(across(everything(), \(x) paste0(na.omit(unique(x)), collapse = "; ")), .by = c("country", "main_activity", "clustered"))
+
+  non_summarised_df |>
+    left_join(summarised_df, by = c("country", "main_activity", "clustered")) |>
+    mutate(across(everything(), \(x) ifelse(!nzchar(x), NA_character_, x))) |>
     distinct() |>
-    mutate(multi_match = case_when(
-      multi_match == "" ~ "FALSE",
-      TRUE ~ multi_match
-    ))
+    mutate(multi_match = ifelse(is.na(.data$multi_match), FALSE, TRUE))
```

Comment:
`tidyr::replace_na()` seems more evocative:

``` r
data <- tibble::tibble(multi_match = c(TRUE, FALSE, NA), x = c(TRUE, FALSE, NA))
data
#> # A tibble: 3 × 2
#>   multi_match x    
#>   <lgl>       <lgl>
#> 1 TRUE        TRUE 
#> 2 FALSE       FALSE
#> 3 NA          NA

tidyr::replace_na(data, list(multi_match = FALSE))
#> # A tibble: 3 × 2
#>   multi_match x    
#>   <lgl>       <lgl>
#> 1 TRUE        TRUE 
#> 2 FALSE       FALSE
#> 3 FALSE       NA
```

<sup>Created on 2023-10-17 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Implemented! Thanks for suggesting improved solution